hyperspy/learn/mva.py

```diff
@@ -1317,22 +1369,679 @@ def _estimate_elbow_position(self, curve_values):
+
+    def _get_number_pca_components_for_clustering(self):
+        if self.learning_results.number_significant_components is None:
+            raise ValueError("Number of pca components not defined, "
+                             "please run decomposition first.")
+        else:
+            number_pca_components = self.learning_results.number_significant_components
+        return number_pca_components
+
+    def _get_number_clusters_for_clustering(self):
+        if self.learning_results.number_of_clusters is None:
+            raise ValueError("Number of clusters not defined, "
+                             "please run evaluate_number_of_clusters first.")
+        else:
+            number_of_clusters = self.learning_results.number_of_clusters
+        return number_of_clusters
+
+    def _cluster_analysis(self,
+                          n_clusters,
+                          scaled_data,
+                          algorithm='kmeans',
+                          **kwargs,
+                          ):
+        """
+        Cluster analysis of a scaled data - internal
+
+        Parameters
+        ----------
+        n_clusters : int
+            Number of clusters to find.
+        scaled_data : numpy array - (number_of_samples,number_of_features)
+        algorithm: string "kmeans" or "agglomerative"
+             See scikit-learn documentation. Default "kmeans"
+        **kwargs
+            Additional parameters passed to the clustering algorithm.
+            This may include `n_init`, the number of times the algorithm is
+            restarted to optimize results.
+
+        """
+        if import_sklearn.sklearn_installed is False:
+            raise ImportError('sklearn is not installed. Nothing done.')
+        if algorithm not in cluster_algorithms:
+            raise ValueError("Cluster algorithm {algorithm} not supported or "
+                             "not correctly defined.")
+
+        alg = cluster_algorithms[algorithm](n_clusters=n_clusters, **kwargs)
+
+        alg.fit(scaled_data)
+
+        return alg
+
+    def cluster_analysis(self,
+                         n_clusters=None,
+                         scaling="minmax",
+                         scaling_kwargs={},
+                         use_decomposition_results=True,
+                         use_decomposition_for_centers=False,
+                         number_pca_components=None,
+                         navigation_mask=None,
+                         signal_mask=None,
+                         algorithm='kmeans',
+                         return_info=False,
+                         **kwargs):
+        """
+        Cluster analysis of a signal
+
+        Results are stored in `learning_results`.
+
+        Parameters
+        ----------
+        n_clusters : int
+            Number of clusters to find.
+        scaling : {"standard","norm","minmax", or scikit learn scaling method}
+            default: 'standard'
```

Comment:
This default value is different to the one on line 1416

Reply:
fixed